addons/binding/org.openhab.binding.elerotransmitterstick/src/main/java/org/openhab/binding/elerotransmitterstick/handler/EleroChannelHandler.java

```diff
@@ -0,0 +1,117 @@
+    protected EleroTransmitterStickHandler bridge;
+
+    public EleroChannelHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        bridge = (EleroTransmitterStickHandler) getBridge().getHandler();
+
+        setChannelIds();
+        for (int channelId : channelIds) {
+            bridge.addStatusListener(channelId, this);
+        }
+
+        if (bridge.getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    protected void setChannelIds() {
+        String channelIdStr = getThing().getProperties().get(PROPERTY_CHANNEL_ID);
+        channelIds = new int[] { Integer.parseInt(channelIdStr) };
+    }
+
+    @Override
+    public void dispose() {
+        for (Integer channelId : channelIds) {
+            bridge.removeStatusListener(channelId, this);
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            logger.debug("Bridge for Elero channel handler for thing {} ({}) changed status to {}",
+                    getThing().getLabel(), getThing().getUID(), bridgeStatusInfo.getStatus().toString());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received command {} for channel {}", command, channelUID);
+
+        if (channelUID.getIdWithoutGroup().equals(CONTROL_CHANNEL)) {
+            if (command == UpDownType.UP) {
+                bridge.getStick().sendCommand(CommandType.UP, channelIds);
+            } else if (command == UpDownType.DOWN) {
+                bridge.getStick().sendCommand(CommandType.DOWN, channelIds);
+            } else if (command == StopMoveType.STOP) {
+                bridge.getStick().sendCommand(CommandType.STOP, channelIds);
+            } else if (command instanceof PercentType) {
+                CommandType cmd = CommandType.getForPercent(((PercentType) command).intValue());
+                if (cmd != null) {
+                    bridge.getStick().sendCommand(cmd, channelIds);
+                } else {
+                    logger.warn("Unhandled command {}.", command);
+                }
+            } else if (command == RefreshType.REFRESH) {
+                bridge.getStick().requestUpdate(channelIds);
+            }
+        }
+    }
+
+    @Override
+    public void statusChanged(int channelId, ResponseStatus status) {
+        logger.debug("Received updated state {} for thing {}", status, getThing().getUID().toString());
+
+        updateState(STATUS_CHANNEL, new StringType(status.toString()));
+
+        if (ResponseStatus.getPercentageFor(status) != -1) {
+            updateState(CONTROL_CHANNEL, new PercentType(ResponseStatus.getPercentageFor(status)));
```

Comment:
Please store the result of `ResponseStatus.getPercentageFor(status)` in a variable if you use it twice.

Reply:
Done